### ISSUE TITLE
Red linting 0.12 code

### ISSUE THREAD

**Issue by sdcscripts** [hashicorp/vscode-terraform#293]
When I use 0.12 syntax I always end up with 'red' error linting that suggests there is a syntactically error with the code? The code is syntactically correct as per 0.12 rules, is this just me or are others also finding this behaviour? 

![image](https://user-images.githubusercontent.com/17180046/76758753-463c7c80-6782-11ea-8914-557ded3796a9.png)

**Comment by tenwit** [comment]
Turn off indexing in your settings. That's for tf 0.11, so that 0.12 syntax is failing and causing the error marks.
```
"terraform.indexing": {
        "enabled": false,
        "liveIndexing": false
}
```

**Comment by sdcscripts** [comment]
Hi, 

Thanks, that seemed to change where the red text lands but now it is underlining references to resources that are found in other .tf files in the directory. For instance, I have a network.tf which references a resource group in nsg.tf in the same directory. It's suggesting the resource group is not there, even though it is. It also thinks the resource is a variable. The configurations all apply fine still.

![image](https://user-images.githubusercontent.com/17180046/76848550-cecb2380-683b-11ea-81c8-725f6e2deec5.png)

Cheers

**Comment by haodeon** [comment]
Have you enabled the language server for 0.12?

**Comment by matt-canty-dragon** [comment]
I have the same issue.

![image](https://user-images.githubusercontent.com/56073682/76955516-42d3fd00-690a-11ea-97f7-b7f85fb1b545.png)

![image](https://user-images.githubusercontent.com/56073682/76955595-6139f880-690a-11ea-808a-f7a57e8add9d.png)

**Comment by dking3876** [comment]
> Have you enabled the language server for 0.12?

How do you enable the language server? nothing seems to be in the readme?

**Comment by matt-canty-dragon** [comment]
@dking3876 only documented in the change log -> https://github.com/mauve/vscode-terraform/blob/master/CHANGELOG.md#140

**Comment by VihanAgarwal97** [comment]
Enabling the language server fixes this issue.

**Comment by shcrouch** [comment]
Hi, 

I have the language server enabled and I still have this issue. It doesn't recognise resources such as resource groups that are defined in adjacent terraform files. The code is syntactically correct and runs just fine, but I just have lots of red everywhere. 

![image](https://user-images.githubusercontent.com/62308630/77016494-afcdae00-696f-11ea-95a3-bfafeeb70b52.png)


![image](https://user-images.githubusercontent.com/62308630/77016326-3b930a80-696f-11ea-9e96-3fe1a80b04e6.png)

![image](https://user-images.githubusercontent.com/62308630/77016343-477ecc80-696f-11ea-9623-8ca9e362c3ef.png)


My settings show language server enabled: 

![image](https://user-images.githubusercontent.com/62308630/77016391-69784f00-696f-11ea-877a-397c2ebd568e.png)

and language server is working: 

![image](https://user-images.githubusercontent.com/62308630/77016435-8b71d180-696f-11ea-8119-e9d4900939cd.png)

**Comment by mwawrusch** [comment]
Same here. Identical settings to the one shown above, language server enabled.

**Comment by jasonrberk** [comment]
bump.....still not working and getting warning / errors in VS

**Comment by codephobia** [comment]
I am getting this issue as well. Seems to only occur as far as I can tell when in a module and the resource or data object is initially created in a different file.

File A: contains resource
File B: references resource in File A with error

If the reference is in the same file it doesn't have an issue finding it (both in File A, or both in File B).

Doesn't seem to be an issue in outputs.tf files though.

**Comment by TechnicallyJoe** [comment]
I have this exact same issue. Language server installed and indexing set to false.

Now i'm getting errors all over my configuration and string interpolation warnings everywhere. However when i change these to the 0.12 way, its still seen as an error.

**Comment by liamfoneill** [comment]
Same issue. My resource group is in a file called "resourcegroup.tf". I have a virtual network in a file called "virtual-network-spoke2.tf". The VNET is deployed into the resource group but because the resource group is in a different file it gives me this syntax error:

![image](https://user-images.githubusercontent.com/1243172/77301883-53102180-6ce8-11ea-838b-a969af15cd49.png)

**Comment by conanza** [comment]
I had been seeing the same issue even with the identical settings as above users until I rebooted vscode and installed/updated the Language Server.

Settings
![image](https://user-images.githubusercontent.com/8183981/77616068-ff6f2500-6eed-11ea-86d8-b8c22361c127.png)

Install/Update Language Server via Command Palette (cmd+shift+P)
<img width="512" alt="Screen Shot 2020-03-25 at 11 12 11 PM" src="https://user-images.githubusercontent.com/8183981/77616172-478e4780-6eee-11ea-86c4-06daec0ee04a.png">
![image](https://user-images.githubusercontent.com/8183981/77616260-74daf580-6eee-11ea-9fd5-e6e4d55acd7e.png)

Displaying various errors correctly
![image](https://user-images.githubusercontent.com/8183981/77616325-99cf6880-6eee-11ea-94ad-607060e91010.png)
![image](https://user-images.githubusercontent.com/8183981/77616329-9b992c00-6eee-11ea-9e14-ba84de6773db.png)

**Comment by sdcscripts** [comment]
Hi, this deals with interpolation warnings, but not the issue as reflected above by users where resource groups are defined in one file and then referred to in another file, that still incorrectly errors as per:

 

> Same issue. My resource group is in a file called "resourcegroup.tf". I have a virtual network in a file called "virtual-network-spoke2.tf". The VNET is deployed into the resource group but because the resource group is in a different file it gives me this syntax error:
> 
> ![image](https://user-images.githubusercontent.com/1243172/77301883-53102180-6ce8-11ea-838b-a969af15cd49.png)
> 
> Paging @juliosueiras as this seems to be a bug in the language server.

**Comment by conanza** [comment]
Works fine across files for me too 🤷‍♂. Perhaps this is specific to the Provider
![image](https://user-images.githubusercontent.com/8183981/77714286-ddc97880-6f95-11ea-9ceb-c704497dedcc.png)

**Comment by ubenmackin** [comment]
I have the same issue on my Windows install or VSCode, but on my OS X install, it seems to traverse files just fine.

All of my code is checked into a git repo, so it is the same across both comptuers (multiple .tf fles). On the Windows VSCode (using just updated language server and VSCode) it throws errors on resources contained in different files. On my OS X install, I don't see the same behaviour, it seems to work fine.

**Comment by shcrouch** [comment]
Is that the case specifically when you define Azure resource groups in one file and refer to them in another?

**Comment by codephobia** [comment]
This isn't related specifically to azure. I was having the issue with aws providers. Am also running on windows vs code.

**Comment by shcrouch** [comment]
Does the error present as "**There is no variable named** ... <provider object name>" ?

**Comment by codephobia** [comment]
@shcrouch Yep. For example: "There is no variable named "aws_subnet"."

**Comment by liamfoneill** [comment]
Just an update (putting here as it seems these issues are related). If I reference a data source (data source in same tf file) the first reference has an error, yet the subsequent ones are red errors:
![image](https://user-images.githubusercontent.com/1243172/77896129-b3b2d780-726f-11ea-931f-f377199412ad.png)

![image](https://user-images.githubusercontent.com/1243172/77896147-bad9e580-726f-11ea-9449-87c2336ee181.png)

**Comment by elreydetoda** [comment]
I had the same error with the packet provider. I then realized I hadn't done `terraform init` in the folder. So, I then did a `terraform init`, disabled the terraform language server, updated the language server, and then enabled it again. After doing that everything worked properly.

**NOTE:** if you get a file is busy error from vscode you can run this command:
```
rm ~/.vscode/extensions/mauve.terraform-1.4.0/lspbin/terraform-lsp{,.tar.gz}
```
got command from [here](https://github.com/mauve/vscode-terraform/issues/254#issuecomment-588296056), but shortened it.

**Comment by sdcscripts** [comment]
Hey @elreydetoda  , (A propósito - me gusta tu nombre de usuario! ;) )  

I tried this but for me it didn't seem to have the same affect, I've tried all kinds of combinations of uninstall \ reinstall \ tried on multiple OS instances, but it's the same every time. 

Thanks anyway

**Comment by elreydetoda** [comment]
(muchas gracias :grin: )

Dang, sorry :cry: . I thought if it worked for me it would work for other.

**Comment by pmalek** [comment]
I'm also getting this for aws provider:

![image](https://user-images.githubusercontent.com/739996/78551274-b7ad9f00-7805-11ea-8754-43abbc5a7945.png)

with the following log:

```
2020-04-06T12:47:50.018+0200 [INFO]  plugin: configuring client automatic mTLS
2020-04-06T12:47:50.063+0200 [DEBUG] plugin: starting plugin: path=/home/USER/.vscode-server/extensions/mauve.terraform-1.4.0/lspbin/terraform-provider-aws_v2.56.0_x4 args=[/home/USER/.vscode-server/extensions/mauve.terraform-1.4.0/lspbin/terraform-provider-aws_v2.56.0_x4]
2020-04-06T12:47:50.065+0200 [DEBUG] plugin: plugin started: path=/home/USER/.vscode-server/extensions/mauve.terraform-1.4.0/lspbin/terraform-provider-aws_v2.56.0_x4 pid=31294
2020-04-06T12:47:50.065+0200 [DEBUG] plugin: waiting for RPC address: path=/home/USER/.vscode-server/extensions/mauve.terraform-1.4.0/lspbin/terraform-provider-aws_v2.56.0_x4
2020-04-06T12:47:50.093+0200 [INFO]  plugin.terraform-provider-aws_v2.56.0_x4: configuring server automatic mTLS: timestamp=2020-04-06T12:47:50.093+0200
2020-04-06T12:47:50.139+0200 [DEBUG] plugin.terraform-provider-aws_v2.56.0_x4: plugin address: address=/tmp/plugin141250327 network=unix timestamp=2020-04-06T12:47:50.139+0200
2020-04-06T12:47:50.139+0200 [DEBUG] plugin: using plugin: version=5
2020-04-06T12:47:50.334+0200 [DEBUG] plugin: plugin process exited: path=/home/USER/.vscode-server/extensions/mauve.terraform-1.4.0/lspbin/terraform-provider-aws_v2.56.0_x4 pid=31294
2020-04-06T12:47:50.334+0200 [DEBUG] plugin: plugin exited
```

**Comment by AndrewNao** [comment]
I'm getting this issue too... I'm using an older provider... azurerm 1.41... could this be related?

**Comment by paultyng** [comment]
We just released [v2.0.0-rc.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) of the extension. The main features include:

* Added syntax support for 0.12
* Added [terraform-ls](https://github.com/hashicorp/terraform-ls) usage by default (currently on 0.3.0, which offers basic provider code completion)

You can find additional information and specifics in the [release notes](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) and [CHANGELOG](https://github.com/hashicorp/vscode-terraform/blob/master/CHANGELOG.md).

With this release we expect that many of the prior issues and PRs are no longer relevant or have been addressed, and are therefore being closed. If you feel the action taken on an issue or PR is in error, please comment as such and we can figure out the appropriate way to address it.

We plan to add the final 2.0.0 release to the marketplace soon, but are actively seeking your feedback now on the release candidates. You can download the .vsix from the releases page and [manually install it in VS Code](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix) to try it out.
